src/types/decodeData.ts

```diff
@@ -3,8 +3,8 @@ import type { EncodeDataType, URLDataWithHash } from './encodeData/JSONURL';
 
 export interface DataInput {
   keyName: string; // can be the name or the hex/hash
-  value;
-  dynamicKeyParts?: string | string[] | number;
+  value: unknown;
```

Comment:
Why `unknown`?

Reply:
Unknown because you don't know whether it's an object or Hex or a number. That way you don't have to do the as any thing.